workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/RealOverlayDialogHolder.kt

```diff
@@ -0,0 +1,21 @@
+package com.squareup.workflow1.ui.container
+
+import android.app.Dialog
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+
+@WorkflowUiExperimentalApi
+internal class RealOverlayDialogHolder<OverlayT : Overlay>(
+  initialEnvironment: ViewEnvironment,
+  override val dialog: Dialog,
+  runner: (rendering: OverlayT, environment: ViewEnvironment) -> Unit
+) : OverlayDialogHolder<OverlayT> {
+
+  private var _environment: ViewEnvironment = initialEnvironment
+  override val environment: ViewEnvironment get() = _environment
+
+  override val runner = { newScreen: OverlayT, newEnvironment: ViewEnvironment ->
+    _environment = newEnvironment
+    runner(newScreen, newEnvironment)
```

Comment:
Is there worthwhile naming that would make it more clear that this `runner` lambda declared on line 19 is calling the `runner` lambda on line 11? Took a second to realize there were two different lambdas named `runner` and we're not intentionally causing a stack overflow lol

Reply:
`realRunner` lol.

Reply:
Oh, that's not good. Will tidy.